### ISSUE TITLE
Failing tests on Windows

### ISSUE THREAD

**Issue by BYK** [shelljs/shelljs#296]
There are failing tests on Windows. My system is Node 4.2.4 with Windows 10:

```
BYK@ ~▶Documents▶Projects▶shelljs⛕ master$ npm test

> shelljs@0.5.3 test C:\Users\BYK\Documents\Projects\shelljs
> node scripts/run-tests

'.' is not recognized as an internal or external command,
operable program or batch file.
*** JSHINT FAILED! (return code != 0)

Running test: cat.js
Running test: cd.js
Running test: chmod.js

assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: 438 == 493
    at Object.<anonymous> (C:\Users\BYK\Documents\Projects\shelljs\test\chmod.js:23:8)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:136:18)
    at node.js:963:3
*** TEST FAILED! (missing exit code "123")

Running test: common.js
Running test: config.js
Running test: cp.js

assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: 0 == 1
    at Object.<anonymous> (C:\Users\BYK\Documents\Projects\shelljs\test\cp.js:145:8)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:136:18)
    at node.js:963:3
*** TEST FAILED! (missing exit code "123")

Running test: dirs.js
Running test: echo.js
Running test: env.js
Running test: exec.js
Running test: find.js
Running test: grep.js
Running test: ln.js

assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: false == true
    at Object.<anonymous> (C:\Users\BYK\Documents\Projects\shelljs\test\ln.js:51:1)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:136:18)
    at node.js:963:3
*** TEST FAILED! (missing exit code "123")

Running test: ls.js
Running test: make.js
Running test: mkdir.js
Running test: mv.js
Running test: popd.js
Running test: pushd.js

assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: 'resources\\pushd' == 'resources/pushd'
    at Object.<anonymous> (C:\Users\BYK\Documents\Projects\shelljs\test\pushd.js:193:8)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:136:18)
    at node.js:963:3
*** TEST FAILED! (missing exit code "123")

Running test: pwd.js
Running test: rm.js
Running test: sed.js
Running test: tempdir.js
Running test: test.js

assert.js:89
  throw new assert.AssertionError({
  ^
AssertionError: false == true
    at Object.<anonymous> (C:\Users\BYK\Documents\Projects\shelljs\test\test.js:76:8)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:136:18)
    at node.js:963:3
*** TEST FAILED! (missing exit code "123")

Running test: to.js
Running test: toEnd.js
Running test: touch.js
Running test: which.js

*******************************************************
WARNING: Some tests did not pass!
*******************************************************
npm ERR! Test failed.  See above for more details.
```

**Comment by BYK** [comment]
Well, this looks like a dupe of #294.

**Comment by nfischer** [comment]
This is still a good issue to keep open. This will help us diagnose the actual errors with the tests.

#295 is a partial fix for #294. It looks like your PR #297 will help with the individual test cases.

The only other thing is to see if we can fix the issues with `exec()` spawning in the home directory instead of current directory for `cmd.exe` users (not sure if this is really even fixable).

**Comment by BYK** [comment]
>The only other thing is to see if we can fix the issues with exec() spawning in the home directory instead of current directory for cmd.exe users (not sure if this is really even fixable).

Can you reference the issue from here too?

**Comment by nfischer** [comment]
@BYK The issue with `exec()` is related to #250. When running a shelljs script from `cmd.exe`, I've seen that any `exec()` calls will run in the home directory.

```Javascript
require('shelljs/global');
echo(pwd()); // This is ShellJS's pwd command, which **will be** correct on Windows
exec('cd'); // Windows 'cd' is the equivalent of Unix's pwd command
```

I'm observing that, given I'm in a directory **other than** my home directory, then this will not output the same string twice (as I would expect it to).

I'm running this on Windows 7 64bit, using `cmd.exe` as my shell. If you see different behavior, with or without a different configuration, let me know.

**Comment by BYK** [comment]
@nfischer I can't reproduce this on my Windows 10, Node 4 installation. Tried both from `cdm.exe` and the shell I'm using (`xonsh`). Both output the directory I started `node` from which is the correct working directory. I simply run the following:

```
$ node
> require('shelljs/global')
{}
> exec('cd')
'the working directory I'm in followed by \r\n'
```

**Comment by nfischer** [comment]
Interesting. I'm running through node v5 on my Windows machine. Perhaps it's Windows 7 specific, or node v5 specific (unlikely), or perhaps it's specific to my machine.

Personally, I don't care too much about running `exec()` on my personal windows machine, but I imagine that if I have the issue, others might as well. I've seen some articles online that mention that `child_process.execSync` calls will spawn in the home directory, so it appears to not be only me (but I can't find the posts right now). I've heard that running as admin can help the issue, but haven't tested this.

Either way, if anyone else can confirm either behavior on their windows system, that would be appreciated.

**Comment by nfischer** [comment]
Fixed in #297
